index.ts

```diff
@@ -1,110 +1,110 @@
+    if (!res.data.pageInfo) return {} as PlayListResponse;
     return {
-        totalResults: res.data.pageInfo.totalResults,
-        videos: res.data.items,
-        nextPageToken: res.data.nextPageToken
-    }
+        totalResults: res.data.pageInfo.totalResults as number,
+        videos: res.data.items as youtube_v3.Schema$PlaylistItem[],
+        nextPageToken: res.data.nextPageToken as string
+    };
 }
 
-async function getAllVideos(nextPageToken: string, items: youtube_v3.Schema$PlaylistItem[], count: number): Promise<string[]> {
-    const res = await getVideos(nextPageToken)
-    if (!res) return
-    const newItems = items.concat(res.videos)
-    console.log('newItems', newItems)
+async function getAllVideos(nextPageToken: string, items: youtube_v3.Schema$PlaylistItem[], count: number)
+        : Promise<youtube_v3.Schema$PlaylistItem[]> {
+    const res = await getVideos(nextPageToken);
+    if (!res) return [];
+    const newItems = items.concat(res.videos);
     if (count < 1) {
         // if (count < res.totalResults / 50) {
-        return getAllVideos(res.nextPageToken = "", newItems, count + 1)
+        return getAllVideos(res.nextPageToken = '', newItems, count + 1);
     } else {
-        return newItems
+        return newItems;
     }
 }
 
-function getRating(review): string {
-    if (review.includes("CLASSIC")) return "CLASSIC/10"
-    if (review.includes("NOT GOOD")) return "NOT GOOD/10"
+function getRating(review: string): string {
+    if (review.includes('CLASSIC')) return 'CLASSIC/10';
+    if (review.includes('NOT GOOD')) return 'NOT GOOD/10';
 
     const regex = /[0-9][10]*\/[1][0]/g;
-    const rating = review.match(regex) || "??/??";
+    const rating = review.match(regex) || '??/??';
     return rating[0];
 }
 
-function getArtist(review): string {
-    if (review.resourceId.videoId === "F-Fd5YG2pWs") return
-    if (review.resourceId.videoId === "MNnibsPJSDY") return
-    if (review.resourceId.videoId === "LDMNhCOs0G0") return
-
-    const regex = /.*(?=-)/g;
-    const artist = review.title.trim().match(regex);
-    return artist[0].replace(/\s*$/, "");
+function getArtist(review: youtube_v3.Schema$PlaylistItemSnippet): string {
+    return getRegexGroupFromTitle(review, /.*(?=-)/g);
 }
 
-function getAlbum(review): string {
-    if (review.resourceId.videoId === "F-Fd5YG2pWs") return
-    if (review.resourceId.videoId === "MNnibsPJSDY") return
-    if (review.resourceId.videoId === "LDMNhCOs0G0") return
-
-    const regex = /(?<=-).*/g;
-    const album = review.title.trim().match(regex)
-    // .replace("ALBUM REVIEW", "")
-    return album[0].replace(/\s*$/, "");
+function getAlbum(review: youtube_v3.Schema$PlaylistItemSnippet): string {
+    return getRegexGroupFromTitle(review, /(?<=-).*/g);
 }
 
-interface Review {
-    date: string,
-    url: string,
-    artist: string,
-    album: string,
-    rating: string
+function isBlacklistedReview(review: youtube_v3.Schema$PlaylistItemSnippet) : boolean {
```

Comment:
This fn name is a bit problematic

Reply:
Would love to see this fn name used in a positive way too. Like if we could showcase Black artists and the associated reviews, `isBlacklistedReview` would make sense in that context.

Reply:
Good call. Renamed method.